subprojects/messaging/src/main/java/org/gradle/internal/serialize/ExceptionPlaceholder.java

```diff
@@ -151,10 +151,23 @@ public Object transform(Object obj) {
         }
 
         Throwable placeholder = new PlaceholderException(type, message, getMessageExec, toString, toStringRuntimeExec, causeThrowable);
-        placeholder.setStackTrace(stackTrace);
+        try {
+            placeholder.setStackTrace(stackTrace);
+        } catch (NullPointerException e) {
+            diagnoseNullPointerException(e);
+        }
         return placeholder;
     }
 
+    private void diagnoseNullPointerException(NullPointerException e) {
+        // https://github.com/gradle/gradle-private/issues/1368
+        StringBuilder sb = new StringBuilder();
+        for (StackTraceElement element : stackTrace) {
+            sb.append(element).append("\n");
+        }
```

Comment:
Have you considered using `Throwable.printStrackTrace(new PrintWriter(new StringWriter()))` instead?

Reply:
I don't quite understand. The problem is that there's a `StackTraceElement[]` which might contains a `null` and I want to print this malicious `StackTraceElement[]`, how can `Throwable.printStrackTrace()` help?